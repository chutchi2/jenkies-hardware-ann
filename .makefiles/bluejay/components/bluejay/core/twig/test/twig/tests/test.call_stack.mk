#-------------------------------------------------------------------------------
# Filename: test.call_stack.mk
#
# Description:
# This file ...
#-------------------------------------------------------------------------------
include ../../unit_test.mk

include ../../../makefiles/twig.call_stack.mk
include ../../../makefiles/twig.stack.mk

all: ; @:

#-------------------------------------------------------------------------------
# Tests
#-------------------------------------------------------------------------------
#-------------------------------------------------------------------------------
$(call BEGIN_TEST,Empty stack)

$(call ut.assert_equal,$(call twig.call_stack.peek),<makefile>)
$(call ut.assert_equal,$(call twig.call_stack.top),<makefile>[1])

#-------------------------------------------------------------------------------
$(call BEGIN_TEST,push/pop)

$(call ut.assert_equal,$(call twig.call_stack.peek),<makefile>)

$(call twig.call_stack.push,test1)
$(call twig.call_stack.push,test2)
$(call twig.call_stack.push,test3)

$(call ut.assert_equal,$(call twig.call_stack.peek),test3)
$(call twig.call_stack.pop)
$(call ut.assert_equal,$(call twig.call_stack.peek),test2)
$(call twig.call_stack.pop)
$(call ut.assert_equal,$(call twig.call_stack.peek),test1)
$(call twig.call_stack.pop)
$(call ut.assert_equal,$(call twig.call_stack.peek),<makefile>)

#-------------------------------------------------------------------------------
$(call BEGIN_TEST,push/pop an empty string)

$(call ut.assert_equal,$(call twig.call_stack.peek),<makefile>)

$(call twig.call_stack.push,  )
$(call ut.assert_equal,$(call twig.call_stack.peek),<makefile>)

#-------------------------------------------------------------------------------
$(call BEGIN_TEST,pop call stack too many times)

$(call ut.assert_equal,$(call twig.call_stack.peek),<makefile>)
$(call twig.call_stack.pop)

#-------------------------------------------------------------------------------
$(call BEGIN_TEST,top)

$(call ut.assert_equal,$(call twig.call_stack.peek),<makefile>)

$(call twig.call_stack.push,test1)
$(call twig.call_stack.push,test1)
$(call twig.call_stack.push,test2)
$(call twig.call_stack.push,test1)
$(call twig.call_stack.push,test3)

$(call ut.assert_equal,$(call twig.call_stack.top),test3[1])
$(call twig.call_stack.pop)
$(call ut.assert_equal,$(call twig.call_stack.top),test1[3])
$(call twig.call_stack.pop)
$(call ut.assert_equal,$(call twig.call_stack.top),test2[1])
$(call twig.call_stack.pop)
$(call ut.assert_equal,$(call twig.call_stack.top),test1[2])
$(call twig.call_stack.pop)
$(call ut.assert_equal,$(call twig.call_stack.top),test1[1])
$(call twig.call_stack.pop)
$(call ut.assert_equal,$(call twig.call_stack.top),<makefile>[1])
