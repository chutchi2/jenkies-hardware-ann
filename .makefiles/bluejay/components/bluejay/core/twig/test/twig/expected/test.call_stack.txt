
========== Test 1 [ Empty stack ]

========== Test 2 [ push/pop ]

========== Test 3 [ push/pop an empty string ]

========== Test 4 [ pop call stack too many times ]
test.call_stack.mk:52: Call stack popped too many times!

========== Test 5 [ top ]
